crates/rpc/src/eth_rpc.rs

```diff
@@ -147,7 +148,16 @@ impl EthApiServer for KakarotEthRpc {
     }
 
     async fn transaction_count(&self, _address: Address, _block_number: Option<BlockId>) -> Result<U256> {
-        Ok(U256::from(3))
+        let starknet_block_id = _block_number.map(ethers_block_id_to_starknet_block_id);
+        let starknet_block_id = match starknet_block_id {
+            Some(Ok(b)) => b,
+            Some(Err(e)) => return Err(e.into()),
+            None => StarknetBlockId::Tag(BlockTag::Latest),
+        };
+
+        let transaction_count = self.kakarot_client.nonce(_address, starknet_block_id).await?;
```

Comment:
nice! logic is fine here, so we'll need to refactor the inner nonce() function -> don't use account_contract function, rather use starknet JSON RPC route getNonce()

Reply:
But, we discussed on the chat right, that we should use the protocol level nonce, which is from the account contract?

Will the nonce of the Account contract deployed for the EOA by Kakarot always be the same as the nonce of the contract in Starknet?

If that is the case, then it makes sense to move to Starknet RPC implementation.

Reply:
> we discussed on the chat, that we should use the protocol level nonce
as far as i understand, this protocol level nonce is called using Starknet RPC client, not Account.get_nonce() 
Account.get_nonce() is a reproduction of the protocol level nonce.

Reply:
Gotcha, as discussed, I have updated to use starknet protocol level nonces.